### PR TITLE
[MIG] account voucher operating unit to v10

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
==============================
Account Voucher Operating Unit
==============================

This module introduces Operating Units to the Account Voucher model. It also
introduces security rules to manage access control only to users that can
operate on the OU of the voucher.

Customer or Supplier Receipts
-----------------------------

* The Operating Unit is assigned to the voucher, and then to the journal
  items when the voucher is posted.

# Depends on https://github.com/OCA/operating-unit/pull/50

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
depends on https://github.com/OCA/operating-unit/pull/45

File: account_voucher_operating_unit/models/account_voucher.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    @api.multi
+    def _get_default_operating_unit(self):
+        user = self.env['res.users'].browse(self._uid)
+        return user.default_operating_unit_id
+
+    operating_unit_id = fields.Many2one(
+        'operating.unit',
+        string='Operating Unit',
+        default=_get_default_operating_unit,
```
Comment (serpentcs-dev1) [inline account_voucher_operating_unit/models/account_voucher.py]: There's already a method for this
```
default=lambda self: self.env['res.users'].operating_unit_default_get(self._uid)
```

File: account_voucher_operating_unit/models/account_voucher.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    )
+
+    @api.multi
+    @api.constrains('operating_unit_id', 'company_id')
+    def _check_company_operating_unit(self):
+        for rec in self:
+            if rec.company_id and rec.operating_unit_id and \
+                    rec.company_id != rec.operating_unit_id.company_id:
+                raise ValidationError(_('The Company in the Move Line and in '
+                                        'the Operating Unit must be the same.'
```
Comment (serpentcs-dev1) [inline account_voucher_operating_unit/models/account_voucher.py]: Should be : The Company in the Voucher and in

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
:+1:

**Review by MiquelRForgeFlow — APPROVED** [review]
yes :+1:

**Pull request merged**